File: framework/ecommerce-framework-api/src/main/java/com/sdl/ecommerce/api/model/ECommerceEclItem.java
package com.sdl.ecommerce.api.model;

import com.sdl.webapp.common.api.model.entity.EclItem;

/**
 * ECommerce ECL Item
 *
 * @author nic
 */
public class ECommerceEclItem extends EclItem {

    /**
     * @return external ID of the ECL item (category or product ID)
     */
    public String getExternalId() {
        if ( this.getExternalMetadata() != null ) {
            return (String) this.getExternalMetadata().get("Id");
        }
        return this.getFileName();
    }

    /**
     * @return name used by the external system, such as category title, product name etc
     */
    public String getExternalName() {
        if ( this.getExternalMetadata() != null ) {
            return (String) this.getExternalMetadata().get("Name");
        }
        return null;
    }
}

File: dxa.net/ecommerce-framework-dotnet/SDL.ECommerce.DXA/Models/ECommerceEclItem.cs
using Sdl.Web.Common.Models;
using SDL.ECommerce.Api.Model;

namespace SDL.ECommerce.DXA.Models
{
    /// <summary>
    /// E-Commerce ECL Item
    /// </summary>
    public class ECommerceEclItem : EclItem
    {
        /// <summary>
        /// External ID
        /// </summary>
        public string ExternalId
        {
            get
            {
                var id = this.GetEclExternalMetadataValue("Id") as string;
                if ( id == null )
                {
                    id = FileName;
                }
                return id;
            }
        }

        /// <summary>
        /// External Name
        /// </summary>
        public string ExternalName
        {
            get
            {
                return this.GetEclExternalMetadataValue("Name") as string;
            }
        }

        /// <summary>
        /// Concrete references to products & categories consumed by ECL views
        /// </summary>
        public IProduct Product { get; set; }
        public ICategory Category { get; set; }
    }
}